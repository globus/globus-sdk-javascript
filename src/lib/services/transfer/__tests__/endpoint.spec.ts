import serviceTestSuite from '../../../../__utils__/service-test-suite';
import { endpoint } from '..';

const ENDPOINT = 'c591c905-2674-4227-9d31-1ff9485945a7';

serviceTestSuite('transfer', 'endpoint', (fetch) => {
  test('get', async () => {
    await endpoint.get(ENDPOINT);
    expect(fetch).toHaveBeenCalled();
    expect(fetch).toHaveBeenCalledWith(
      `https://transfer.api.globusonline.org/v0.10/endpoint/${ENDPOINT}`,
      {
        headers: {},
      },
    );
  });
});
