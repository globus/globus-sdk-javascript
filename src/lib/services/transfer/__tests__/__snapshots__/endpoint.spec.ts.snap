// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`endpoint create 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "content-length": "146",
    "content-type": "application/json",
    "host": "transfer.api.globusonline.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "json": {
    "DATA_TYPE": "shared_endpoint",
    "display_name": "my GCP guest collection",
    "host_endpoint_id": "c591c905-2674-4227-9d31-1ff9485945a7",
    "host_path": "/",
  },
  "method": "POST",
  "url": "https://transfer.api.globusonline.org/v0.10/shared_endpoint",
}
`;

exports[`endpoint delete 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "host": "transfer.api.globusonline.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "method": "DELETE",
  "url": "https://transfer.api.globusonline.org/v0.10/endpoint/c591c905-2674-4227-9d31-1ff9485945a7",
}
`;

exports[`endpoint get 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "host": "transfer.api.globusonline.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "method": "GET",
  "url": "https://transfer.api.globusonline.org/v0.10/endpoint/c591c905-2674-4227-9d31-1ff9485945a7",
}
`;

exports[`endpoint update 1`] = `
{
  "headers": {
    "accept": "*/*",
    "accept-encoding": "gzip,deflate",
    "connection": "close",
    "content-length": "67",
    "content-type": "application/json",
    "host": "transfer.api.globusonline.org",
    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
  },
  "json": {
    "DATA_TYPE": "endpoint",
    "description": "This is my GCP collection.",
  },
  "method": "PUT",
  "url": "https://transfer.api.globusonline.org/v0.10/endpoint/c591c905-2674-4227-9d31-1ff9485945a7",
}
`;
